src/dioptra/restapi/v1/users/schema.py

```diff
@@ -37,3 +44,102 @@ class UserRefSchema(Schema):
         metadata=dict(description="URL for accessing the full User resource."),
         relative=True,
     )
+
+
+class UserMutableFieldsSchema(Schema):
+    """The schema for the mutable data fields in a User resource."""
+
+    username = fields.String(
+        attribute="name", metadata=dict(description="Username of the User.")
+    )
+
+    email = fields.String(
+        attribute="email", metadata=dict(description="Email of the User.")
+    )
+
+
+class UserSchema(UserMutableFieldsSchema):
+    """The schema for a User that is not the logged-in User."""
+
+    password = fields.String(
+        attribute="password",
+        metadata=dict(description="Password for the User resource."),
+        load_only=True,
+    )
+
+    id = fields.Integer(
+        attribute="id",
+        metadata=dict(description="ID for the User resource."),
+        dump_only=True,
+    )
+
+
+class UserCurrentSchema(UserSchema):
+    """The schema for the currently logged-in User."""
+
+    memberOf = fields.Nested(
+        GroupRefSchema,
+        attribute="member_of",
+        metadata=dict(description="A list of Groups the User is a part of."),
+        many=True,
+        dump_only=True,
+    )
+    createdOn = fields.DateTime(
+        attribute="created_on",
+        metadata=dict(
+            description="Timestamp when the User resource was created.",
+            dump_only=True,
+        ),
+    )
+    lastModifiedOn = fields.DateTime(
+        attribute="last_modified_on",
+        metadata=dict(
+            description="Timestamp when the User resource was last modified."
+        ),
+        dump_only=True,
+    )
+    lastLoginOn = fields.DateTime(
+        attribute="last_login_on",
+        metadata=dict(
+            description="Timestamp when the User resource was last logged in."
+        ),
+        dump_only=True,
+    )
+    passwordExpiresOn = fields.DateTime(
+        attribute="password_expires_on",
+        metadata=dict(
+            description="Timestamp when the User resource password expires on."
+        ),
+        dump_only=True,
+    )
+
+
+class UserPasswordSchema(Schema):
+    """The schema for changing a User's password."""
+
+    oldPassword = fields.String(
+        attribute="old_password",
+        metadata=dict(description="Old password for the User resource."),
+    )
+
+    newPassword = fields.String(
+        attribute="new_password",
+        metadata=dict(description="New password for the User resource."),
+    )
+
```

Comment:
Also needs a `confirmNewPassword` field for initial user registration, so its `load_only=True`.

https://github.com/usnistgov/dioptra/blob/60af0a6aac3af7e4e9a47e16afa0e6877af9f05f/src/dioptra/restapi/v0/user/schema.py#L49-L55